src/main/java/org/broadinstitute/consent/http/authentication/OAuthAuthenticator.java

```diff
@@ -67,42 +71,47 @@ private AuthUser buildAuthUserFromHeaders(Map<String, String> headers) {
    * Attempt to get the registration status of the current user and set the value on AuthUser
    *
    * @param authUser The AuthUser
-   * @return A cloned AuthUser with Sam registration status
+   * @return A cloned AuthUser with Sam registration status and a Consent User
    */
   private AuthUser getUserWithStatusInfo(AuthUser authUser) {
     if (authUser == null || authUser.getEmail() == null) {
       logWarn("AuthUser/email is null, cannot get user status info");
       return null;
     }
+    try {
+      authUser.setUser(userService.findUserByEmail(authUser.getEmail()));
+    } catch (Exception e) {
+      logException("Error finding Consent user: " + authUser.getEmail(), e);
+    }
     try {
       UserStatusInfo userStatusInfo = samService.getRegistrationInfo(authUser);
       if (Objects.nonNull(userStatusInfo)) {
         // safety check in case the call to generic user (i.e. Google) failed.
-        if (Objects.isNull(authUser.getEmail())) {
+        if (authUser.getEmail() == null) {
           authUser.setEmail(userStatusInfo.getUserEmail());
         }
-        if (Objects.isNull(authUser.getName())) {
+        if (authUser.getName() == null) {
           authUser.setName(userStatusInfo.getUserEmail());
         }
       } else {
         logWarn("Error getting user status info back from Sam for user: " + authUser.getEmail());
       }
       return authUser.deepCopy().setUserStatusInfo(userStatusInfo);
     } catch (NotFoundException e) {
-      Gson gson = new Gson();
       try {
         // Try to post the user to Sam if they have not registered previously
         UserStatus userStatus = samService.postRegistrationInfo(authUser);
-        if (Objects.nonNull(userStatus) && Objects.nonNull(userStatus.getUserInfo())) {
+        if ((userStatus != null) && (userStatus.getUserInfo() != null)) {
           authUser.setEmail(userStatus.getUserInfo().getUserEmail());
         } else {
+          Gson gson = GsonUtil.gsonBuilderWithAdapters().create();
           logWarn("Error posting to Sam, AuthUser not able to be registered: " + gson.toJson(authUser));
```

Comment:
Do we need to use `gson` for the purposes of logging `authuser`? Is this a logging message we will need to look at, and does it have information we wouldn't already have in the logs from the ingress proxy?

Reply:
Gson, in this context, necessary for the current log method to work without changes. At present, I don't know if proxy logs have enough information to debug this issue.

Reply:
What I'm asking here is, do we need to use `gson` to extract the information needed for logging? We have the object itself, can we instead log the important fields directly without first converting to JSON? Or is the JSON output needed in the logs for some other reason?

Reply:
We certainly could log this differently than we're doing now with selected information. I don't remember the history behind why we're logging the full object. The only reason for gson here would have been to do just that though, log the full object.